src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerOptions.cs

```diff
@@ -40,10 +50,34 @@ public class ExceptionHandlerOptions
     public bool AllowStatusCode404Response { get; set; }
 
     /// <summary>
-    /// Gets or sets a delegate used to map an exception to a http status code.
+    /// Gets or sets a delegate used to map an exception to an HTTP status code.
     /// </summary>
     /// <remarks>
     /// If <see cref="StatusCodeSelector"/> is <c>null</c>, the default exception status code 500 is used.
     /// </remarks>
     public Func<Exception, int>? StatusCodeSelector { get; set; }
+
+    /// <summary>
+    /// Gets or sets a callback that can return <see langword="true" /> be used to suppress diagnostics by <see cref="ExceptionHandlerMiddleware" />.
+    /// The default value is to suppress diagnostics if the exception was handled by an <see cref="IExceptionHandler"/> service instance registered in the DI container.
```

Comment:
We should probably call out how the behavior changes when we make this null instead of the default value. Or it might be even better to leave this null by default and move the default `ExceptionHandlerService` check into ExceptionHandlerMiddlewareImpl.